lib/cloudex/cloudinary_api.ex

```diff
@@ -164,16 +181,16 @@ defmodule Cloudex.CloudinaryApi do
     }"
   end
 
-  @spec post(tuple | String.t(), binary, map, integer()) ::
+  @spec post(tuple | String.t(), binary, map, keyword) ::
           {:ok, %Cloudex.UploadedImage{}} | {:ok, %Cloudex.UploadedVideo{}} | {:error, any}
-  defp post(body, source, opts, recv_timeout) do
-    with {:ok, raw_response} <- common_post(body, opts, recv_timeout),
+  defp post(body, source, opts, poison_opts) do
+    with {:ok, raw_response} <- common_post(body, opts, poison_opts),
          {:ok, response} <- @json_library.decode(raw_response.body),
          do: handle_response(response, source)
   end
 
-  defp common_post(body, opts, recv_timeout) do
-    HTTPoison.request(:post, url_for(opts), body, @cloudinary_headers, credentials(recv_timeout))
+  defp common_post(body, opts, poison_opts) do
+    HTTPoison.request(:post, url_for(opts), body, @cloudinary_headers, credentials() ++ poison_opts)
```

Comment:
Do we want to do this for deleting as well? Like have the option to pass in poison_opts?

Reply:
So based on the comments above, I think opts should still contain the `:httpoison` key here, might want to double check by inspecting, so we can call `poison_opts = get_httpoison(opts)` here and not have to pass it in as an argument